web/src/app/components/duration.utils.ts

```diff
@@ -21,32 +21,31 @@ export interface DurationItem {
   endNsec: number|Long;    // nsec time of the last state
 }
 
-const NANO_TO_DAYS = (10 ** 9) * 60 * 60 * 24;
-const NANO_TO_HOURS: number = (10 ** 9) * 60 * 60;
-const NANO_TO_MINUTES: number = (10 ** 9) * 60;
 const NANO_TO_SECONDS: number = (10 ** 9);
+const NANO_TO_MINUTES: number = (10 ** 9) * 60;
+const NANO_TO_HOURS: number = (10 ** 9) * 60 * 60;
+const NANO_TO_DAYS = (10 ** 9) * 60 * 60 * 24;
+const NANO_TO_WEEKS = (10 ** 9) * 60 * 60 * 24 * 7;
 
 export const UNIT_CONVERSION: {[unit: string]: number} = {
   seconds: NANO_TO_SECONDS,
   minutes: NANO_TO_MINUTES,
   hours: NANO_TO_HOURS,
-  days: NANO_TO_DAYS
+  days: NANO_TO_DAYS,
+  weeks: NANO_TO_WEEKS
 };
 
 /**
  * Finds the unit of time that best describes the majority of the durations of
  * the pushInfos.
  *
  * @param pushInfos Array of pushes for a single push def
- * @return one of [seconds, minutes, hours, days]
+ * @return one of [seconds, minutes, hours, days, weeks]
  */
 export function findDurationUnit(pushInfos: step189_2020.IPushInfo[]): string {
-  const unitCounter: {[unit: string]: number} = {
-    seconds: 0,
-    minutes: 0,
-    hours: 0,
-    days: 0
-  };
+  const unitCounter:
+      {[unit: string]:
+           number} = {seconds: 0, minutes: 0, hours: 0, days: 0, weeks: 0};
```

Comment:
Can we keep the original formatting?

Reply:
the clang formatter doesn't let me but Razvan mentioned it worked for him?

Reply:
The formatter seem use the long format if there is a `,` at the end. 

Here is a form that works:

```
  const unitCounter: {[unit: string]: number} = {
    seconds: 0,
    minutes: 0,
    hours: 0,
    days: 0,
    weeks: 0,
  };
```